### PR TITLE
[assert] Catch up to node types

### PR DESCRIPTION
**Pull request by jablko**
Incorporate existing changes to node types.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/assert/ts3.6/index.d.ts [hunk 1]
```diff
@@ -1,55 +1,77 @@
...
+    function doesNotReject(
+        block: (() => Promise<any>) | Promise<any>,
+        error: RegExp | Function,
+        message?: string | Error,
+    ): Promise<void>;
+
+    function match(value: string, regExp: RegExp, message?: string | Error): void;
+    function doesNotMatch(value: string, regExp: RegExp, message?: string | Error): void;
+
     const strict: typeof assert;
```
Comment (ExE-Boss) [inline types/assert/ts3.6/index.d.ts:75]:
```suggestion
    const strict: Omit<
        typeof assert,
        | 'strict'
        | 'deepEqual'
        | 'notDeepEqual'
        | 'equal'
        | 'notEqual'
    > & {
        (value: any, message?: string | Error): void;
        strict: typeof strict;
        deepEqual: typeof deepStrictEqual;
        notDeepEqual: typeof notDeepStrictEqual;
        equal: typeof strictEqual;
        notEqual: typeof notStrictEqual;
    };
```

File: types/assert/ts3.6/tslint.json [hunk 2]
```diff
@@ -1 +1,6 @@
-{ "extends": "dtslint/dt.json" }
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "ban-types": false
+    }
+}
```
Comment (ExE-Boss) [inline types/assert/ts3.6/tslint.json:6]:
```suggestion
{ "extends": "dtslint/dt.json" }
```

File: types/assert/tslint.json [hunk 3]
```diff
@@ -1 +1,6 @@
-{ "extends": "dtslint/dt.json" }
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "ban-types": false
+    }
+}
```
Comment (ExE-Boss) [inline types/assert/tslint.json:6]:
```suggestion
{ "extends": "dtslint/dt.json" }
```

**Review by ExE-Boss — COMMENTED** [review]
If some lines require `ban‑types` to be disabled, then use the form:

```ts
// tslint:disable-next-line: ban-types
```

File: types/assert/index.d.ts [hunk 4]
```diff
@@ -5,63 +5,103 @@
...
+        (value: any, message?: string | Error): asserts value;
+        strict: typeof strict;
+        deepEqual: typeof deepStrictEqual;
+        notDeepEqual: typeof notDeepStrictEqual;
+        equal: typeof strictEqual;
+        notEqual: typeof notStrictEqual;
+        ok(value: any, message?: string | Error): asserts value;
+        strictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+        deepStrictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+        ifError(value: any): asserts value is null | undefined;
```
Comment (ExE-Boss) [inline types/assert/index.d.ts:103]: See <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49469>:

```suggestion

        // Mapped types and assertion functions are incompatible?
        // TS2775: Assertions require every name in the call target
        // to be declared with an explicit type annotation.
        ok: typeof ok;
        strictEqual: typeof strictEqual;
        deepStrictEqual: typeof deepStrictEqual;
        ifError: typeof ifError;
```

**Comment by ExE-Boss** [comment]
Files in the `types/node/` directory shouldn’t be edited in this PR.

File: types/assert/ts3.6/tslint.json (hunk 2 above)
Comment (jablko) [reply types/assert/ts3.6/tslint.json]: Done :heavy_check_mark:

File: types/assert/tslint.json (hunk 3 above)
Comment (jablko) [reply types/assert/tslint.json]: Done :heavy_check_mark:

File: types/assert/index.d.ts (hunk 4 above)
Comment (jablko) [reply types/assert/index.d.ts]: Done :heavy_check_mark:

File: types/assert/ts3.6/index.d.ts (hunk 1 above)
Comment (jablko) [reply types/assert/ts3.6/index.d.ts:77]: Good suggestion but `Omit<>` was added in 3.5 and these types are still for Minimum TypeScript Version < 3.5?

**Comment by jablko** [comment]
Yup, once #49469 and #49486 land those extra commits/diffs will be absorbed :+1:

**Comment by orta** [comment]
Now that both of the upstream PRs are good - are we in a spot where this PR can be merged?

**Comment by jablko** [comment]
Yes I think so :+1:

**Comment by orta** [comment]
OK, node reviewers, last chance to give this a one over - when I see it again I'm merging 👍🏻

**Comment by andrewbranch** [comment]
@orta ready?

**Comment by andrewbranch** [comment]
> Files in the types/node/ directory shouldn’t be edited in this PR.

Is this still true? There are still edits to node typings shown. Need a rebase/merge, or am I missing something, or is GitHub lying?

**Comment by jablko** [comment]
@andrewbranch I guess GitHub shows the differences from the merge base, not HEAD. I rebased and it shows just the `types/assert` edits/commits now :heavy_check_mark:

**Comment by orta** [comment]
Alright, well, this classes as me seeing it again and I think we'll take it. 

This PR is a good step forwards and doesn't look like it should cause massive breaking changes.

**Pull request merged**